pkg/nsdns/manager_test.go

```diff
@@ -0,0 +1,154 @@
+package nsdns
+
+import (
+	"os"
+	"testing"
+)
+
+import (
+	"github.com/stretchr/testify/assert"
+	apinetworkingv1 "k8s.io/api/networking/v1"
+)
+
+import (
+	"github.com/Eagerod/kube-namesilo-dns/pkg/namesilo_api"
+)
+
+type MockNamesiloApi struct{}
+
+func (nsapi MockNamesiloApi) ListDNSRecords() ([]namesilo_api.ResourceRecord, error) {
+	return []namesilo_api.ResourceRecord{}, nil
+}
+
+func (nsapi MockNamesiloApi) UpdateDNSRecord(rr namesilo_api.ResourceRecord) error {
+	return nil
+}
+
+func (nsapi MockNamesiloApi) AddDNSRecord(rr namesilo_api.ResourceRecord) error {
+	return nil
+}
+
+func (nsapi MockNamesiloApi) DeleteDNSRecord(rr namesilo_api.ResourceRecord) error {
+	return nil
+}
+
+func TestNewDnsManager(t *testing.T) {
+	ov := os.Getenv("NAMESILO_API_KEY")
+	os.Setenv("NAMESILO_API_KEY", "a")
+
+	dm, err := NewDnsManager("a", "b")
+	assert.NoError(t, err)
+	assert.Equal(t, "a", dm.BareDomainName)
+	assert.Equal(t, "b", dm.TargetIngressClass)
+
+	os.Setenv("NAMESILO_API_KEY", "")
+	_, err = NewDnsManager("a", "b")
+	assert.Equal(t, "failed to find NAMESILO_API_KEY in environment; cannot proceed", err.Error())
+
+	os.Setenv("NAMESILO_API_KEY", ov)
+}
+
+func TestNewDnsManagerWithApiKey(t *testing.T) {
+	dm, err := NewDnsManagerWithApiKey("a", "b", "c")
+	assert.NoError(t, err)
+	assert.Equal(t, "a", dm.BareDomainName)
+	assert.Equal(t, "b", dm.TargetIngressClass)
+
+	dm, err = NewDnsManagerWithApiKey("", "b", "c")
+	assert.Nil(t, dm)
+	assert.Equal(t, "must provide a domain name to target DNS record updates", err.Error())
+
+	dm, err = NewDnsManagerWithApiKey("a", "", "c")
+	assert.Nil(t, dm)
+	assert.Equal(t, "must provide an ingress class to generate DNS records", err.Error())
+}
+
+func TestShouldProcessIngress(t *testing.T) {
+	dm, err := NewDnsManagerWithApiKey("a", "b", "c")
+	assert.NoError(t, err)
+
+	ingress := apinetworkingv1.Ingress{}
+	ingress.Annotations = map[string]string{}
+	ingress.Annotations["kubernetes.io/ingress.class"] = dm.TargetIngressClass
+
+	assert.True(t, dm.ShouldProcessIngress(&ingress))
+
+	ingress.Annotations["kubernetes.io/ingress.class"] = dm.TargetIngressClass + "not"
+
+	assert.False(t, dm.ShouldProcessIngress(&ingress))
+}
+
+// This technically only provides coverage, without actually testing much.
```

Comment:
This is the main thing I care about here.

Lines are _covered_, but nothing really fails if all the behaviour goes jank. Should have a way of confirming that the right API calls are made.